examples/tv-casting-app/android/App/app/src/main/java/com/chip/casting/app/CommissionerDiscoveryFragment.java

```diff
@@ -8,143 +8,183 @@
+        Callback callback = (Callback) this.getActivity();
+        View.OnClickListener manualCommissioningButtonOnClickListener =
+            v -> callback.handleCommissioningButtonClicked(null);
+        manualCommissioningButton.setOnClickListener(manualCommissioningButtonOnClickListener);
+        ArrayAdapter<DiscoveredNodeData> arrayAdapter = new VideoPlayerCommissionerAdapter(getActivity(), commissionablePlayerList);
+        final ListView list = getActivity().findViewById(R.id.commissionerList);
+        list.setAdapter(arrayAdapter);
+        list.setOnItemClickListener((parent, view1, position, id) -> {
+            DiscoveredNodeData discoveredNodeData = (DiscoveredNodeData)parent.getItemAtPosition(position);
+            Log.d(
+                    TAG,
+                    "CommissionerResolveListener.onServiceResolved.OnClickListener.onClick called for "
+                    + discoveredNodeData);
+            Callback callback1 =
+                (Callback) getActivity();
+            callback1.handleCommissioningButtonClicked(discoveredNodeData);
         });
 
-    Log.d(TAG, "Auto discovering");
-    tvCastingApp.discoverVideoPlayerCommissioners(
-        DISCOVERY_DURATION_SECS, successCallback, failureCallback);
-  }
-
-  @VisibleForTesting
-  public String getCommissionerButtonText(DiscoveredNodeData commissioner) {
-    String main = commissioner.getDeviceName() != null ? commissioner.getDeviceName() : "";
-    String aux =
-        "" + (commissioner.getProductId() > 0 ? "Product ID: " + commissioner.getProductId() : "");
-    aux +=
-        commissioner.getDeviceType() > 0
+        this.successCallback =
+            new SuccessCallback<DiscoveredNodeData>() {
+                @Override
+                public void handle(DiscoveredNodeData discoveredNodeData) {
+                    Log.d(TAG, "Discovered a Video Player Commissioner: " + discoveredNodeData);
+                    new Handler(Looper.getMainLooper())
+                        .post(() -> {
+                            if (commissionablePlayerList.contains(discoveredNodeData)) {
+                                Log.d(TAG, "Replacing existing entry in players list");
+                                arrayAdapter.remove(discoveredNodeData);
+                            }
+                            arrayAdapter.add(discoveredNodeData);
+                        });
+                }
+            };
+
+        this.failureCallback =
+            new FailureCallback() {
+                @Override
+                public void handle(MatterError matterError) {
+                    Log.e(TAG, "Error occurred during video player commissioner discovery: " + matterError);
+                    if (MatterError.ERROR_SERVICE_LOST == matterError.getErrorCode()) {
+                        Log.d(TAG, "Attempting to restart service");
+                        tvCastingApp.discoverVideoPlayerCommissioners(successCallback, this);
```

Comment:
It looks like ERROR_SERVICE_LOST would be returned if a device on the network, dropped off. Why are we restarting the discovery on the casting app in that event?

Reply:
I saw that event occurring when the name change event was sent, this caused us to lose additional events that happened after the fact. Restarting the service seemed to help resolve this.